tests/trainer/test_trainer.py

```diff
@@ -764,7 +764,7 @@ def training_step(self, batch, batch_idx, optimizer_idx=None):
         terminate_on_nan=True,
     )
 
-    with pytest.raises(ValueError, match=r".*The loss returned in `training_step` is nan or inf.*"):
+    with pytest.raises(ValueError, match=r".*The loss returned in `training_step` is.*"):
```

Comment:
@Borda - raising a RuntimeError runs into the issues here: https://github.com/PyTorchLightning/pytorch-lightning/issues/6842

The trainer catches it so the exception is never raised